### DISCUSSION TITLE
Nodes virtualization

### DISCUSSION THREAD

**Discussion by Aemulatio** [facebook/lexical#5994] [Q&A]
Does anybody did this? 
Why I need it? So, because in our project we creating a lot of side nodes (we need them) and in this case when the symbol counter reach 30k lags become more and more obvious. But as we expect that our users' devices will be not that powerful as ours and they will be suffer from this problem

**Comment by ivailop7** [comment]
30K nodes is quite a lot. Reading should be probably fine, but editing with that many nodes would be probably a very sub-par experience. Lazy loading is achivable, ImageNode for example is lazy loaded in the playground. Virtualization probably too, but I'd imagine editing won't be improved much because of the update handlers logic that still needs to run on the entire editor state tree, irrespective if in view or not. That's my read on it.
